crates/workspace/src/pane.rs

```diff
@@ -2647,6 +2677,54 @@ mod tests {
         });
     }
 
+    #[gpui::test]
+    async fn test_add_item_capped_to_max_tabs(cx: &mut TestAppContext) {
+        init_test(cx);
+        let fs = FakeFs::new(cx.executor());
+
+        let project = Project::test(fs, None, cx).await;
+        let (workspace, cx) = cx.add_window_view(|cx| Workspace::test_new(project.clone(), cx));
+        let pane = workspace.update(cx, |workspace, _| workspace.active_pane().clone());
+
+        for i in 0..7 {
+            add_labeled_item(&pane, format!("{}", i).as_str(), false, cx);
+        }
+        set_max_tabs(cx, Some(5));
+        add_labeled_item(&pane, "7", false, cx);
```

Comment:
Currently the max_tab algorithm only runs when a new tab is created, hence this test order.

Maybe we could consider removing tabs when max_tabs is set. I don't know that we have these kind of hooks in the codebase, and I'm not sure it is that important (I feel like it would be set once in while as a global setting by a user anyway).